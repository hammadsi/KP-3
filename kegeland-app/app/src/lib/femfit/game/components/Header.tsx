import React from 'react';
import {Image, StyleSheet, View} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {Text, useTheme} from 'react-native-paper';

import CoinSprite from '~assets/femfit/gold.png';
import Icon from '~components/Icon';

import constants from '../constants';
const {HEADER_HEIGHT, COIN_SIZE} = constants;

export type HeaderProps = {
  score: number;
  toggle: () => void;
};

/**
 * Component for rendering the game header panel
 * @param param0 the props
 * @see {@link HeaderProps}
 */
const Header: React.FC<HeaderProps> = ({score, toggle}) => {
  const {colors} = useTheme();
  return (
    <View
      style={[
        styles.header,
        {height: HEADER_HEIGHT, backgroundColor: colors.surface},
      ]}>
      <View style={styles.scoreContainer}>
        <Text style={styles.score}>{score}</Text>
        <Image
          style={styles.scoreSprite}
          source={CoinSprite}
          resizeMode="stretch"
        />
      </View>
      <View>
        <TouchableOpacity onPress={toggle}>
          <Icon icon="menu" color={colors.primary} size={28} />
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    zIndex: 1000,
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 8,
  },
  scoreContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  score: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  scoreSprite: {
    marginLeft: 8,
    width: COIN_SIZE,
    height: COIN_SIZE,
  },
});

export default Header;
