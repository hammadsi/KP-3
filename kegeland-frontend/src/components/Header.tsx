import { Box, Flex, Icon, useMediaQuery, Text, HStack } from '@chakra-ui/react';
import { MdMenu, MdChevronLeft } from 'react-icons/md';
import { useNavigate } from 'react-router-dom';

import useBreadcrumbs from '../hooks/useBreadcrumbs';

import Breadcrumbs from './Breadcrumbs';

type HeaderProps = {
  isSidebarOpen: boolean;
  toggleSidebar: () => void;
};

const Header: React.FC<HeaderProps> = ({ toggleSidebar, isSidebarOpen }) => {
  const navigate = useNavigate();
  const { crumbs } = useBreadcrumbs();
  const [isGreaterThanLg] = useMediaQuery('(min-width: 62em)');
  const moveBtn = isSidebarOpen && !isGreaterThanLg;
  return (
    <Box
      display="flex"
      paddingX={4}
      width={'100%'}
      height={'60px'}
      flexDirection="row"
      alignItems="center"
      bgColor="white"
      boxShadow="md"
      zIndex={1000}>
      <Flex
        flexDir="row"
        zIndex={999}
        overflow="hidden"
        marginX="auto"
        w="100%"
        maxW="8xl"
        justify="space-between"
        align="stretch">
        <Flex justify="center" align="center">
          <Icon
            as={MdMenu}
            onClick={toggleSidebar}
            position={moveBtn ? 'absolute' : 'relative'}
            left={moveBtn ? '300px' : 0}
            color="primary.600"
            transform="all 300ms"
            _hover={{ cursor: 'pointer', color: 'primary.700' }}
            aria-label="Toggle sidebar"
            width="40px"
            height="40px"
          />
          {isGreaterThanLg && crumbs.length > 0 && (
            <Breadcrumbs marginLeft={4} crumbs={crumbs} />
          )}
        </Flex>
        <Flex justify="end" flex={1} flexGrow={1}>
          {crumbs.length > 0 && (
            <HStack
              spacing={0}
              onClick={() => navigate(-1)}
              color="primary.600"
              transform="color 300ms"
              _hover={{ cursor: 'pointer', color: 'primary.700' }}>
              <Icon
                as={MdChevronLeft}
                aria-label="Toggle sidebar"
                width="24px"
                height="24px"
              />
              <Text fontWeight="semibold">Go back</Text>
            </HStack>
          )}
        </Flex>
      </Flex>
    </Box>
  );
};

export default Header;
