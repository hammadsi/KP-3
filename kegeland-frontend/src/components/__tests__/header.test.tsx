import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Header from '../Header';

describe('Test header', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      value: () => {
        return {
          matches: false,
          addListener: () => {},
          removeListener: () => {},
        };
      },
    });
  });

  it('renders correctly closed', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header isSidebarOpen={false} toggleSidebar={() => {}} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly open', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header isSidebarOpen={true} toggleSidebar={() => {}} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
