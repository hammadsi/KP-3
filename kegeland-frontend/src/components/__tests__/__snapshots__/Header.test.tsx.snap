// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test header renders correctly closed 1`] = `
<div
  className="css-1li01za"
>
  <div
    className="css-96p7xp"
  >
    <div
      className="css-gmuwbf"
    >
      <svg
        aria-label="Toggle sidebar"
        className="chakra-icon css-1lkjvgo"
        fill="currentColor"
        focusable={false}
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </div>
    <div
      className="css-1wzaiav"
    />
  </div>
</div>
`;

exports[`Test header renders correctly open 1`] = `
<div
  className="css-1li01za"
>
  <div
    className="css-96p7xp"
  >
    <div
      className="css-gmuwbf"
    >
      <svg
        aria-label="Toggle sidebar"
        className="chakra-icon css-na4eyy"
        fill="currentColor"
        focusable={false}
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </div>
    <div
      className="css-1wzaiav"
    />
  </div>
</div>
`;
